web/thesauruses/c/17/functions.json

```diff
@@ -0,0 +1,90 @@
+            "name": "Function that returns a value and takes no parameters"
+        },
+        "return_value_function_with_parameters": {
+            "code": "int foo(int x, int y){ return x + y; }",
+            "comment": "Return value is converted to return type if a conversion is available.",
+            "name": "Function that returns a value and takes 1 or more defined parameters"
+        },
+        "return_value_function_with_keyword_parameters": {
+            "code": "int sum(int a=5,int b=7)",
+            "comment":"default parameters",
+            "name": "Function that returns a value and that takes 0 or more defined keyword parameters"
+        },
+        "return_value_function_variable_parameters": {
+            "code": "eg : double average(int count, ...)",
+            "comment": "a header file <stdargs.h> is available, these are called variadic functions and count variable is passed with no. of arguments",
+            "name": "Function that returns a value and takes an unknown number of parameters"
+        },
+        "anonymous_function_no_parameters": {
+            "not-implemented": "true",
+            "name": "Anonymous function that takes no parameters"
+        },
+        "anonymous_function_with_parameters": {
+            "not-implemented": "true",
+            "name": "Anonymous function that takes 1 or more defined parameters"
+        },
+        "anonymous_function_with_keyword_parameters": {
+            "not-implemented": "true",
+            "name": "Anonymous function that takes 0 or more defined keyword parameters"
+        },
+        "anonymous_function_variable_parameters": {
+            "not-implemented": "true",
+            "name": "Anonymous function that takes an unknown number of parameters"
+        },
+        "anonymous_function_no_parameters_with_return":{
+            "not-implemented":"true",
+            "name":"Anonymous function that takes no parameters and returns a value"
+        },
+        "anonymous_function_with_parameters_with_return": {
+            "not-implemented":"true",
+            "name":"Anonymous function that takes 1 or more defined parameters and returns a value"
+        },
+        "anonymous_function_with_keyword_parameters_with_return":{
+            "not-implemented":"true",
+            "name":"Anonymous function that takes 0 or more defined keyword parameters and returns a value"
+        },
+        "anonymous_function_variable_parameters_with_return":{
+            "not-implemented":"true",
+            "name":"Anonymous function that takes an unknown number of parameters and returns a value"
+         }, 
+        "call_subroutine": {
+            "code": "foo()",
+            "comment": "Calls the function 'foo' - must be defined first.",
+            "name": "Call subroutine"
+        },
+        "return_from_subroutine": {
+            "code": "return",
+            "name": "Return from subroutine"
+        }
```

Comment:
Subroutines are like temporary jumps in code. They're different. I'm guessing these are "not-implemented".